djangocms_bootstrap4/helpers.py

```diff
@@ -32,3 +32,30 @@ def get_plugin_template(instance, prefix, name, templates):
 # otherwise they will not be added to /locale/
 # https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#other-uses-of-lazy-in-delayed-translations
 mark_safe_lazy = lazy(mark_safe, str)
+
+
+# get first element of "choices" (can be nested)
+def get_first_choice(choices):
+    for value, verbose in choices:
+        if not isinstance(verbose, (tuple, list)):
+            return value
+        else:
+            first = get_first_choice(verbose)
+            if first is not None:
+                return first
+    return None
+
+
+# get verbose text of element matching given value in "choices" (can be nested)
+def get_choices_match(choices, value_to_match):
+    for value, verbose in choices:
+        if not isinstance(verbose, (tuple, list)):
+            if value == value_to_match:
+                return verbose
+        else:
+            first = get_first_choice(verbose)
```

Comment:
Might this have to be `return get_choices_match(verbose, value_to_match)`? The following 4 lines then would not be necessary if I see this correctly.

Reply:
Yup, I did need that change. Code is updated. _I was only able to remove two of the four lines previously below that line._

<details>

I kept two because I had to avoid returning prematurely if return value from `get_choices_match` was None—not checking this caused a second element (nested tuple in choices) to never be read. Variables are also less oddly named, now.

</details>